### ISSUE TITLE
unable to generate MSVS solution file via bootstrap

### ISSUE THREAD

**Issue by shiftkey** [electron/electron#9614]
I'm unable to generate the Visual Studio solution file using the [documented steps](https://electron.atom.io/docs/development/build-instructions-windows/). This was previously working in my local development environment around about the 1.6.7 release when I was contributing some work, so I'm working through what's changed in between. Note that removing the `--msvs` flag seems to complete the bootstrap step, and I can `script\build` from there.

* Electron version: current master, e.g. 7d994f2a9
* Operating system: Windows 10 with the relevant tools

### Expected behavior

`python script\bootstrap.py -v --msvs` should complete, which then gives the ability to open an `electron.sln` file in VS for debugging purposes.

### Actual behavior

The tail output (as far back as `cmd` would let me go) can be found [here](https://gist.github.com/shiftkey/9ed42efa0e490b2d24c2da9e72629c8e) but these are the specific errors that I could glean:

```
LINK : fatal error LNK1181: cannot open input file 'C:\Users\shiftkey\src\electron\dist\node-v1.7.2\Release\iojs.lib' [C:\Users\shiftkey\src\electron\spec\node_modules\ref\build\binding.vc
xproj]
Done Building Project "C:\Users\shiftkey\src\electron\spec\node_modules\ref\build\binding.vcxproj" (default targets) -- FAILED.

Done Building Project "C:\Users\shiftkey\src\electron\spec\node_modules\ref\build\binding.sln" (default targets) -- FAILED.


Build FAILED.

"C:\Users\shiftkey\src\electron\spec\node_modules\ref\build\binding.sln" (default target) (1) ->
"C:\Users\shiftkey\src\electron\spec\node_modules\ref\build\binding.vcxproj" (default target) (2) ->
(Link target) ->
  LINK : fatal error LNK1181: cannot open input file 'C:\Users\shiftkey\src\electron\dist\node-v1.7.2\Release\iojs.lib' [C:\Users\shiftkey\src\electron\spec\node_modules\ref\build\binding.
vcxproj]

    0 Warning(s)
    1 Error(s)
```

```
Link:
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64\link.exe /ERRORREPORT:QUEUE /OUT:"C:\Users\shiftkey\src\electron\spec\node_modules\runas\build\Release\runas.node" /INCRE
  MENTAL:NO /NOLOGO kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib oleaut32.lib uuid.lib odbc32.lib DelayImp.lib "C:\Users\shiftkey\src\electron\dist\node-v1.7.2\
  Release\iojs.lib" ole32.lib shell32.lib Delayimp.lib /DELAYLOAD:iojs.exe /DELAYLOAD:node.exe /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /DEBUG /PDB:"C:\U
  sers\shiftkey\src\electron\spec\node_modules\runas\build\Release\runas.pdb" /MAP /MAPINFO:EXPORTS /OPT:REF /OPT:ICF /LTCG /TLBID:1 /DYNAMICBASE /NXCOMPAT /MACHINE:X64 /ignore:4199 /DLL R
  elease\obj\runas\main.obj
  Release\obj\runas\runas_win.obj
  Release\obj\runas\win_delay_load_hook.obj
LINK : fatal error LNK1181: cannot open input file 'C:\Users\shiftkey\src\electron\dist\node-v1.7.2\Release\iojs.lib' [C:\Users\shiftkey\src\electron\spec\node_modules\runas\build\runas.vc
xproj]
Done Building Project "C:\Users\shiftkey\src\electron\spec\node_modules\runas\build\runas.vcxproj" (default targets) -- FAILED.

Done Building Project "C:\Users\shiftkey\src\electron\spec\node_modules\runas\build\binding.sln" (default targets) -- FAILED.


Build FAILED.

"C:\Users\shiftkey\src\electron\spec\node_modules\runas\build\binding.sln" (default target) (1) ->
"C:\Users\shiftkey\src\electron\spec\node_modules\runas\build\runas.vcxproj" (default target) (2) ->
(Link target) ->
  LINK : fatal error LNK1181: cannot open input file 'C:\Users\shiftkey\src\electron\dist\node-v1.7.2\Release\iojs.lib' [C:\Users\shiftkey\src\electron\spec\node_modules\runas\build\runas.
vcxproj]

    0 Warning(s)
    1 Error(s)
```

### How to reproduce

After cloning down `electron/electron` I run these steps:

```
C:\Users\shiftkey\src\electron>npm --version
4.2.0

C:\Users\shiftkey\src\electron>node -v
v7.8.0

C:\Users\shiftkey\src\desktop>npm config list
; cli configs
scope = ""
user-agent = "npm/4.2.0 node/v7.8.0 win32 x64"

; userconfig C:\Users\shiftkey\.npmrc
msvs_version = "2015"

; builtin config undefined
prefix = "C:\\Users\\shiftkey\\AppData\\Roaming\\npm"

; node bin location = C:\Program Files\nodejs\node.exe
; cwd = C:\Users\shiftkey\src\desktop
; HOME = C:\Users\shiftkey
; "npm config ls -l" to show all defaults.

C:\Users\shiftkey\src\electron>python --version
Python 2.7.13

C:\Users\shiftkey\src\electron>python script\bootstrap.py -v --msvs
```

**Comment by MarshallOfSound** [comment]
> LINK : fatal error LNK1181: cannot open input file 'C:\Users\shiftkey\src\electron\dist\node-v1.7.2\Release\iojs.lib

I've had that error recently, I had to roll back my local node version to v6 and then it started working.  Maybe try that out?

**Comment by shiftkey** [comment]
@MarshallOfSound I tried that with `v6.10.3` but the same errors occur.

**Comment by kevinsawicki** [comment]
@shiftkey can you try latest master? I think this may have been addressed by #9589

**Comment by shiftkey** [comment]
@kevinsawicki aside from the warning messages in `update.py` this seems to be fine:

```
...
C:\Python27\python.exe C:\Users\shiftkey\src\electron\script\update.py --msvs
Download directxsdk-x64     1806291  [100.0%]
Warning: Missing input files:
vendor\node\src\tracing\trace_event.hsrc\tree.h
Warning: Missing input files:
vendor\node\src\tracing\trace_event.hsrc\tree.h
```

And I have a working `electron.sln` file at the root of the repository now, so I'm gonna close this out. Thanks for the update!
